test/functional/api/cas/statistics.py

```diff
@@ -0,0 +1,643 @@
+
+
+class RequestStatsChunk:
+    def __init__(self, hits, part_misses, full_misses, total):
+        self.hits = hits
+        self.part_misses = part_misses
+        self.full_misses = full_misses
+        self.total = total
+
+    def __str__(self):
+        return (
+            f"\nHits: {self.hits}\n"
+            f"Partial misses: {self.part_misses}\n"
+            f"Full misses: {self.full_misses}\n"
+            f"Total: {self.total}"
+        )
+
+    def __eq__(self, other):
+        if not other:
+            return False
+        return (
+            self.hits == other.hits
+            and self.part_misses == other.part_misses
+            and self.full_misses == other.full_misses
+            and self.total == other.total
+        )
+
+
+class BlockStats:
+    def __init__(
+        self,
+        core_reads,
+        core_writes,
+        core_total,
+        cache_reads,
+        cache_writes,
+        cache_total,
+        exp_obj_reads,
+        exp_obj_writes,
+        exp_obj_total,
+    ):
+        self.core = BasicStatsChunk(core_reads, core_writes, core_total)
+        self.cache = BasicStatsChunk(cache_reads, cache_writes, cache_total)
+        self.exp_obj = BasicStatsChunk(exp_obj_reads, exp_obj_writes, exp_obj_total)
+
+    def __str__(self):
+        return (
+            f"Block stats:\n"
+            f"Core(s): {self.core}\n"
+            f"Cache: {self.cache}\n"
+            f"Exported object(s): {self.exp_obj}"
+        )
+
+    def __eq__(self, other):
+        if not other:
+            return False
+        return (
+            self.core == other.core
+            and self.cache == other.cache
+            and self.exp_obj == other.exp_obj
+        )
+
+
+class ErrorStats:
+    def __init__(
+        self,
+        cache_read_errors,
+        cache_write_errors,
+        cache_total_errors,
+        core_read_errors,
+        core_write_errors,
+        core_total_errors,
+        total_errors,
+    ):
+        self.cache = BasicStatsChunk(
+            cache_read_errors, cache_write_errors, cache_total_errors
+        )
+        self.core = BasicStatsChunk(
+            core_read_errors, core_write_errors, core_total_errors
+        )
+        self.total_errors = total_errors
+
+    def __str__(self):
+        return (
+            f"Error stats:\n"
+            f"Cache errors: {self.cache}\n"
+            f"Core errors: {self.core}\n"
+            f"Total errors: {self.total_errors}"
+        )
+
+    def __eq__(self, other):
+        if not other:
+            return False
+        return (
+            self.cache == other.cache
+            and self.core == other.core
+            and self.total_errors == other.total_errors
+        )
+
+
+class BasicStatsChunk:
+    def __init__(self, reads, writes, total):
+        self.reads = reads
+        self.writes = writes
+        self.total = total
+
+    def __str__(self):
+        return f"\nReads: {self.reads}\nWrites: {self.writes}\nTotal: {self.total}"
```

Comment:
I would remove \n at the beginning of string, and leave it to be printed by caller.